File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        /*
            pww
        */
        // if(s.length()==0) return 0;
        int l = 0;
        int r = 0;
        vector<int> arr(250,0);
        int ans = 0;
        while(r<s.length()){
            if(arr[s[r]]>0){
                while(s[l]!=s[r]){
                    arr[s[l]]--;
                    l++;
                }
                arr[s[l]]--;
                l++;
                arr[s[r]]++;
            }else{
                arr[s[r]]++;
            }
            // cout<<l<<" "<<r<<endl;
            ans = max(ans,r-l+1);
            r++;
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = 0
        cnt: dict[str, int] = defaultdict(int)
        longest = 0
        for r, c in enumerate(s):
            while(cnt[c]):
                cnt[s[l]] -= 1
                l += 1
            cnt[c] += 1
            longest = max(longest, r - l + 1)
        return longest